src/bl_install.c

```diff
@@ -238,6 +249,107 @@ static int bootloader_send(FILE *fp, int length, u32* pCrc32)
 	return ERR_UNK;
 }
 
+/**
+ * @brief Compare CRCs of the flash memory and the firmware image.
+ * @param fp Firmware file that is supposed to be flashed in the brick.
+ * @param starting_sector First 64 KiB sector of the flash memory to check (0-based)
+ * @param num_sectors Number of 64 KiB sectors to check
+ * @param verbose If true, CRC is checked for each sector individually.
+ *                Otherwise, it is computed across all sectors at once.
+ * @retval error according to enum #ERR
+ */
+int bootloader_crc(FILE *fp, u32 starting_sector, u32 num_sectors, bool verbose)
+{
+	u8 *firmware = calloc(FLASH_SIZE, 1);
+	int read_bytes = fread(firmware, 1, FLASH_SIZE, fp);
```

Comment:
Should check these for error return.

Reply:
I was indeed a bit lazy, I tried to fix this by checking for `ferror()` and by modifying the returned byte count condition.